blockchain/tx_pool_test.go

```diff
@@ -1897,59 +1909,367 @@ func TestDynamicFeeTransactionVeryHighValues(t *testing.T) {
+//
+//	pool, key := setupTxPoolWithConfig(eip1559Config)
+//	defer pool.Stop()
+//
+//	testAddBalance(pool, crypto.PubkeyToAddress(key.PublicKey), big.NewInt(1000000))
+//
+//	tx := dynamicFeeTx(0, 21000, big.NewInt(1), big.NewInt(1), key)
+//	if err := pool.AddRemote(tx); err != nil {
+//		t.Error("error", "got", err)
+//	}
+//
+//	tx2 := dynamicFeeTx(1, 21000, big.NewInt(1), big.NewInt(1), key)
+//	if err := pool.AddRemote(tx2); err != nil {
+//		t.Error("error", "got", err)
+//	}
+//}
+
+// TestDynamicFeeTransactionNotAcceptedNotEnableHardfork tests that the pool didn't accept dynamic tx if the pool didn't enable eip1559 hardfork.
+func TestDynamicFeeTransactionNotAcceptedNotEnableHardfork(t *testing.T) {
 	t.Parallel()
 
-	pool, key := setupTxPoolWithConfig(eip1559Config)
-	defer pool.Stop()
+	pool, key := setupTxPool()
 
-	// Ensure gasFeeCap is greater than or equal to gasTipCap.
-	tx := dynamicFeeTx(0, 100, big.NewInt(1), big.NewInt(2), key)
-	if err := pool.AddRemote(tx); err != ErrTipAboveFeeCap {
-		t.Error("expected", ErrTipAboveFeeCap, "got", err)
+	tx := dynamicFeeTx(0, 21000, big.NewInt(1), big.NewInt(1), key)
+	if err := pool.AddRemote(tx); err != ErrTxTypeNotSupported {
+		t.Error("expected", ErrTxTypeNotSupported, "got", err)
 	}
 
-	// The GasTipCap is equal to gasPrice that config at TxPool.
-	tx2 := dynamicFeeTx(0, 100, big.NewInt(2), big.NewInt(2), key)
-	if err := pool.AddRemote(tx2); err != ErrInvalidGasTipCap {
-		t.Error("expected", ErrInvalidGasTipCap, "got", err)
+	if err := pool.AddLocal(tx); err != ErrTxTypeNotSupported {
+		t.Error("expected", ErrTxTypeNotSupported, "got", err)
 	}
 }
 
+// TestDynamicFeeTransactionAccepted tests that pool accept the transaction which has gasFeeCap bigger than or equal to baseFee.
 func TestDynamicFeeTransactionAccepted(t *testing.T) {
 	t.Parallel()
+	baseFee := big.NewInt(30)
 
 	pool, key := setupTxPoolWithConfig(eip1559Config)
 	defer pool.Stop()
+	pool.SetGasPrice(baseFee)
 
-	testAddBalance(pool, crypto.PubkeyToAddress(key.PublicKey), big.NewInt(1000000))
+	testAddBalance(pool, crypto.PubkeyToAddress(key.PublicKey), big.NewInt(10000000000))
 
-	tx := dynamicFeeTx(0, 21000, big.NewInt(1), big.NewInt(1), key)
+	// The GasFeeCap equal to baseFee and gasTipCap is lower than baseFee(ignored).
+	tx := dynamicFeeTx(0, 21000, big.NewInt(30), big.NewInt(1), key)
 	if err := pool.AddRemote(tx); err != nil {
 		t.Error("error", "got", err)
 	}
 
-	tx2 := dynamicFeeTx(1, 21000, big.NewInt(1), big.NewInt(1), key)
+	// The GasFeeCap bigger than baseFee and gasTipCap is equal to baseFee(ignored).
+	tx2 := dynamicFeeTx(1, 21000, big.NewInt(40), big.NewInt(30), key)
 	if err := pool.AddRemote(tx2); err != nil {
 		t.Error("error", "got", err)
 	}
+
+	// The GasFeeCap greater than baseFee and gasTipCap is bigger than baseFee(ignored).
+	tx3 := dynamicFeeTx(2, 21000, big.NewInt(50), big.NewInt(50), key)
+	if err := pool.AddRemote(tx3); err != nil {
+		t.Error("error", "got", err)
+	}
```

Comment:
Refer to below.
```
fees := []*big.Int{big.NewInt(30), big.NewInt(40), big.NewInt(50)}
tips := []*big.Int{big.NewInt(1), big.NewInt(30), big.NewInt(50)}
for i := 0; i < len(fees); i++ {
    tx := dynamicFeeTx(uint64(i), 21000, fees[i], tips[i], key)
    if err := pool.AddRemote(tx); err != nil {
        t.Error("error", "got", err)
    }
}
```

The redundant code (sequential paste) makes readers read a lot of stuff, even not easy to fix or extend.

Reply:
If you do it the way you suggested, the code will be concise, but I think it will reduce readability. 
I still don't think this code needs to use a loop, and I think it's more intuitive to write it like this for others to figure out the code.

Reply:
If the code is as concise as possible, the readability increase. For example, if the readers figure the what loops are doing out, the only initialization part is the only one to be a key implication of the test. Otherwise like your current change, the readers must see all the code at all times.